Modix.Services/Promotions/PromotionsService.cs

```diff
@@ -327,7 +332,14 @@ public async Task<PromotionCampaignDetails> GetCampaignDetailsAsync(long campaig
         {
             AuthorizationService.RequireClaims(AuthorizationClaim.PromotionsRead);
 
-            return await PromotionCampaignRepository.ReadDetailsAsync(campaignId);
+            var result = await PromotionCampaignRepository.ReadDetailsAsync(campaignId);
+
+            if (result.Subject.Id == AuthorizationService.CurrentUserId)
+            {
+                throw new InvalidOperationException("You can't view comments on your own campaign.");
```

Comment:
The log channel would be reserved for Staff as of this change, so that shouldn't be an issue. This would also make it more viable for us to allow anyone to view campaigns, instead of limiting it to associate+ as it is currently.